MIGRATION.md

```diff
@@ -0,0 +1,69 @@
+# Migration guide
```

Comment:
@poveden This is an attempt at making things a little bit easier for the user. Please feel free to point me at missing things or unclear comments.